server/routerlicious/packages/lambdas/src/deli/lambdaFactory.ts

```diff
@@ -137,7 +157,17 @@ export class DeliLambdaFactory extends EventEmitter implements IPartitionLambdaF
             // The producer as well it shouldn't take. Maybe it just gives an output stream?
             this.forwardProducer,
             this.reverseProducer,
-            this.serviceConfiguration);
+            this.serviceConfiguration,
+            sessionMetric,
+            sessionStartMetric);
+    }
+
+    private logSessionFailureMetrics(
+        sessionMetric: Lumber<LumberEventName.SessionResult> | undefined,
```

Comment:
Curious, any reason we are doing
```
sessionMetric: Lumber<LumberEventName.SessionResult> | undefined,
```

instead of 

```
sessionMetric?: Lumber<LumberEventName.SessionResult>,
```

Does that make the user to explicitly pass `undefined` if one of the metric objects is not available?

Reply:
Oh it must be because we have the errMsg parameter in the last place, so it doesn't allow the first 2 to be optional, likely?

Reply:
Yes. That is correct.